include/unifex/complete_adapt_stream.hpp

```diff
@@ -0,0 +1,52 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <unifex/stream_concepts.hpp>
+
+#include <functional>
+#include <type_traits>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _complete_adapt {
+template <typename Stream, typename AdaptFunc>
+struct _stream {
+  struct type;
+};
+template <typename Stream, typename AdaptFunc>
+using stream =
+    typename _stream<remove_cvref_t<Stream>, remove_cvref_t<AdaptFunc>>::type;
```

Comment:
The AdaptFunc should likely be decay'd here instead of just removing cvref -- similar to the [`adapt_stream` algorithm](https://github.com/facebookexperimental/libunifex/blob/2547d87fb98a1fe70eccc170a8c91b144648517d/include/unifex/adapt_stream.hpp#L33-L35)

Reply:
if there is a consensus on the suggested change, I would add it to next_adapt_stream as well.

Reply:
Looks to me like we _need_ `std::decay_t` here to support `AdaptFunc` being deduced as a function rather than a function pointer: https://godbolt.org/z/eMqW8x93b.

The error in that Godbolt goes away if you manually take the address of `foo`: https://godbolt.org/z/YzE3zvPYn

I expect to be able to construct a function pointer by just naming the function whose address I want (i.e. I don't expect to have to write the `&` in `&foo`) so I think this is a real ergonomics problem.